src/jit/hwintrinsicxarch.cpp

```diff
@@ -445,16 +616,36 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
     {
         return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
     }
-    else if (category == HW_Category_IMM)
+    // Avoid checking stacktop for 0-op intrinsics
+    if (sig->numArgs > 0 && isImmHWIntrinsic(intrinsic, impStackTop().val))
     {
         GenTree* lastOp = impStackTop().val;
-        if (!lastOp->IsCnsIntOrI() && !mustExpand)
+        // The imm-HWintrinsics that do not accept all imm8 values may throw exception or transform to other
+        // intrinsics when the imm argument is not in the valid range
+        if ((flags & HW_Flag_FullRangeIMM) == 0)
```

Comment:
Can you log a bug tracking us adding a helper for checking if a flag is set or unset?

Reply:
Will do, and this comment should be updated, thanks.